lib/spack/spack/relocate.py

```diff
@@ -66,20 +66,20 @@ def get_patchelf():
     Returns the full patchelf binary path.
     """
     # as we may need patchelf, find out where it is
-    if str(spack.architecture.platform()) == 'test':
-        return None
-    if str(spack.architecture.platform()) == 'darwin':
-        return None
     patchelf = spack.util.executable.which('patchelf')
-    if patchelf is None:
+    if patchelf is not None:
+        return patchelf.path
+    else:
+        if str(spack.architecture.platform()) == 'test':
+            return None
+        if str(spack.architecture.platform()) == 'darwin':
+            return None
         patchelf_spec = spack.cmd.parse_specs("patchelf", concretize=True)[0]
         patchelf = spack.repo.get(patchelf_spec)
         if not patchelf.installed:
             patchelf.do_install(use_cache=False)
```

Comment:
Is there a reason this won't use the cache?  We've had users ask why they always have to build patchelf even if it's in the cache

Reply:
Having patchelf patch itself seemed like a bad idea at the time.

Reply:
I better solution might be to download the executable but how do you guarantee libc compatibility.

Reply:
Having packages with only system libraries unpack without requiring patchelf would require extra metadata that is not recorded yet. Maybe a find like standalone=True in the buildinfo dictionary.

Reply:
@tgamblin I propose that I change this to an Exception if patchelf is not in $PATH and require that the user provide patchelf.